### ISSUE TITLE
View reusing bug in the AllControls sample?

### ISSUE THREAD

**Issue by TimLariviere** [fsprojects/Elmish.XamarinForms#136]
**Applies to:** Android & iOS (devices and simulators)

**Description:**
It seems there's a bug with the way how the view is reused between updates.
It happens only after navigating to the "NavigationPage with push/pop" of the AllControls sample.

For a split second, we can see `TabbedPage #2`, `TabbedPage #3` and `CarouselPage` instead of the buttons `Push Page A`, `Push Page B` and `Main page` of the NavigationPage sample.

When clicking `Main page` to go back, same thing. For a split second, there's still the text from the previous page.

But this time those 3 buttons also have a wrong HorizontalOptions value, despite the code being not different than the other buttons.

https://github.com/fsprojects/Elmish.XamarinForms/blob/master/Samples/AllControls/AllControls/AllControls.fs#L222

**Repro:**
1. Start the AllControls app
2. Click "NavigationPage with push/pop"
3. Click "Main page"

<img src="https://user-images.githubusercontent.com/6429007/43153455-a6524c02-8f71-11e8-8103-9d178d3a46da.png" alt="allcontrols-bug-android" height="500" /><img src="https://user-images.githubusercontent.com/6429007/43153456-a6680920-8f71-11e8-93f3-8b682133cf03.png" alt="allcontrols-bug-android" height="500" />

**Comment by dsyme** [comment]
Yes, I noticed this once too, it's great to have it documented.  The view elements must be being recycled, but the differential update is thinking that the horizontal layout settings don't need to be reapplied.  It's odd.  It might be related to assumptions about the default values for the settings.

**Comment by dsyme** [comment]
@TimLariviere Did we ever get to the bottom of this one? Thanks

**Comment by TimLariviere** [comment]
No, I didn't take time to inspect it.
